test/src/test/java/jenkins/agents/WebSocketAgentsTest.java

```diff
@@ -122,4 +124,18 @@ public String call() {
         }
     }
 
+    private static class FatTask extends SlaveToMasterCallable<String, RuntimeException> {
```

Comment:
This should suffice to reproduce on its own (both directions), without the `dd`, right?

Reply:
Yes, though it doesn't seem to exercize the same code path, as the `dd` was working even before I did any change.

Reply:
Possibly due to line-oriented output? Try excluding newline characters using `tr` or whatever.

Reply:
For freestyle, output seems to be buffered in 8k blocks, due to usage of https://github.com/jenkinsci/jenkins/blob/a21f83fb2547a29ee46606405aac9c2624b850a3/core/src/main/java/hudson/util/StreamCopyThread.java#L58

Reply:
Ah OK. So some remote operations involving lots of data would be affected, others would not, and there is no way to tell which other than reading the code or inserting debug traces.